src/Hintman/Webhook.hs

```diff
@@ -5,37 +5,110 @@ module Hintman.Webhook
        ( hintmanApp
        ) where
 
-import GitHub.Data.Webhooks.Events (PullRequestEvent (..))
+import GitHub.Data.Webhooks.Events (InstallationEvent (..), InstallationEventAction (..),
+                                    InstallationRepoEventAction (..),
+                                    InstallationRepositoriesEvent (..), PullRequestEvent (..))
+import GitHub.Data.Webhooks.Payload (HookInstallation (..), HookRepositorySimple (..),
+                                     HookUser (..))
 import Servant (Application, Context ((:.), EmptyContext), Server, hoistServerWithContext,
                 serveWithContext)
 import Servant.API.Generic ((:-), ToServantApi, toServant)
-import Servant.GitHub.Webhook (GitHubEvent, GitHubKey, GitHubSignedReqBody, RepoWebhookEvent (..))
+import Servant.GitHub.Webhook (GitHubEvent, GitHubSignedReqBody, RepoWebhookEvent (..))
 import Servant.Server.Generic (AsServerT)
 
-import Hintman.App (App, Env (..), runAppAsHandler)
+import Hintman.App (App, Env (..), Has, grab, runAppAsHandler)
+import Hintman.Core.Key (GitHubKey)
+import Hintman.Core.PrInfo (Owner (..), Repo (..))
+import Hintman.Core.Token (AppInfo, InstallationAccessToken (..), InstallationId (..))
+import Hintman.Effect.TokenStorage (MonadTokenStorage (..))
+import Hintman.Installation (createAccessToken)
 
 
-newtype HintmanSite route = HintmanSite
-    { hintmanListenRoute :: route
+data HintmanSite route = HintmanSite
+    { -- | Main endpoint that listens PR events.
+      hintmanListenRoute :: route
         :- GitHubEvent '[ 'WebhookPullRequestEvent ]
         :> GitHubSignedReqBody '[JSON] PullRequestEvent
         :> Post '[JSON] ()
+
+      -- | Triggered when a GitHub App has been installed or uninstalled.
```

Comment:
What GitHub App is meant here? Is it about Hintman itself only?

Reply:
Yes, it's about Hintman App. This event happens when you are trying to install Hintman via GitHub web interface. I can rename `GitHub App` to `Hintman GitHub App` or `Hintman` for readability